fedimint-client/src/db.rs

```diff
@@ -416,20 +428,45 @@ pub fn get_core_client_database_migrations() -> BTreeMap<DatabaseVersion, CoreMi
                 return Ok(());
             };
 
-            let global = GlobalClientConfig {
+            let global = GlobalClientConfigV1 {
                 api_endpoints: config_v0.global.api_endpoints,
                 broadcast_public_keys: None,
                 consensus_version: config_v0.global.consensus_version,
                 meta: config_v0.global.meta,
             };
 
-            let config = ClientConfig {
+            let config = ClientConfigV1 {
                 global,
                 modules: config_v0.modules,
             };
 
             dbtx.remove_entry(&id).await;
-            dbtx.insert_new_entry(&ClientConfigKey, &config).await;
+            dbtx.insert_new_entry(&ClientConfigKeyV1, &config).await;
+            Ok(())
+        })
+    });
+
+    migrations.insert(DatabaseVersion(1), |dbtx| {
+        Box::pin(async {
+            let Some(config_v1) = dbtx.remove_entry(&ClientConfigKeyV1).await else {
+                return Ok(());
+            };
+
+            let config = ClientConfig {
+                global: GlobalClientConfig {
+                    api_endpoints: config_v1.global.api_endpoints,
+                    broadcast_public_keys: config_v1.global.broadcast_public_keys,
+                    transaction_fee: Amount::from_msats(0),
+                    consensus_version: config_v1.global.consensus_version,
+                    meta: config_v1.global.meta,
+                },
+                modules: config_v1.modules,
+            };
+
+            dbtx.insert_entry(&ClientConfigKey, &config)
+                .await
+                .expect("We removed the client config v1");
+
```

Comment:
I think in retrospect it was a mistake to patch in `broadcast_public_keys` into the client config. As discussed with @dpc we should add new config as API endpoints and sync it to the client. This has the additional benefit of allowing us to update values by re-fetching it from time to time.